src/yourdfpy/urdf.py

```diff
@@ -379,9 +379,15 @@ def filename_handler_magic(fname, dir):
     Returns:
         str: The file name that exists or the input if nothing is found.
     """
+    relative_handlers = [partial(filename_handler_relative, dir=dir)]
+    if fname.startswith("package://"):
+        package_dir = fname[len("package://"):].split('/')[0]
+        relative_handlers.append(
+            partial(filename_handler_relative, dir=package_dir)
+        )
     return filename_handler_meta(
         fname=fname,
-        filename_handlers=[partial(filename_handler_relative, dir=dir)]
```

Comment:
Sorry for being annoying :-) 
But IMO the cleaner way would be to just add the `filename_handler_ignore_directive`:
```
return filename_handler_meta(
        fname=fname,
        filename_handlers=[
            partial(filename_handler_relative, dir=dir),
            filename_handler_ignore_directive,
        ]
        + _create_filename_handlers_to_urdf_file_recursive(urdf_fname=dir),
    )
```

Reply:
This is constructive, not annoying, please keep it up :smiley: 

I didn't take the time `filename_handler_*` functions beyond what was needed for my use case, so the initial change set was short-sighted. I tested this directive in 1e38e4017708d7dc5c41ebb466a6898078d4d765 and it works just the same with less changes :+1: